File: leetcode/209.min-sub-array-len/cpp/minsubarraylen_test.cpp
#define DOCTEST_CONFIG_IMPLEMENT_WITH_MAIN
#include "doctest.h"

#include "minsubarraylen.cpp"

struct testCase
{
    std::vector<int> nums;
    int target;
    int want;
};

TEST_CASE("MinSubArrayLen")
{
    std::vector<testCase> tcs{
        {std::vector<int>{2, 3, 1, 2, 4, 3}, 7, 2},
        {std::vector<int>{1, 4, 4}, 4, 1},
        {std::vector<int>{1, 1, 1, 1, 1, 1, 1, 1}, 11, 0},
        {std::vector<int>{}, 0, 0},
    };

    Solution s;

    for (auto tc : tcs)
    {
        int got = s.minSubArrayLen(tc.target, tc.nums);
        CHECK(got == tc.want);
    }
}

File: leetcode/209.min-sub-array-len/ts/minsubarraylen.test.ts
import { expect, test } from "vitest";
import minSubArrayLen from "./minsubarraylen.js";

interface testcase {
    nums: number[]
    target: number
    want: number
}

test("minsubarraylen", () => {
    const tcs: testcase[] = [
        { nums: [2, 3, 1, 2, 4, 3], target: 7, want: 2 },
        { nums: [1, 4, 4], target: 4, want: 1 },
        { nums: [1, 1, 1, 1, 1, 1, 1, 1], target: 11, want: 0 },
        { nums: [], target: 0, want: 0 },
    ]

    for (const tc of tcs) {
        expect(minSubArrayLen(tc.target, tc.nums)).toEqual(tc.want)
    }
})
